drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -20,6 +20,8 @@ def perform(document, role = nil)
     all_ids = check_ids(document)
     # Check references
     check_references(document, all_ids)
+    # Check mutually exclusive attributes ("one of" on conditional objects)
+    check_mutually_exclusive_attributes(document)
```

Comment:
@ttoomey I'm not able to request your review, but can you take a look at the changes I made to this file? I wasn't able to express it in JSON.

Reply:
I think you can express this in JSON schema with something like the diff below. However the validation messages may not be as nice, you'd get something to like "Schema error on item 'foo': property '/item/1/bounds/0' is invalid: error type=one of is invalid" 

If that seems okay, I could take a crack at representing the checks in `check_mutually_exclusive_attributes` in the schema, it seems like it wouldn't be too hard. Or we could just stick with the validation function you have implemented here.


```diff
@@ -101,9 +101,14 @@
     "bounds": {
       "type": "object",
       "additionalProperties": false,
-      "required": [
-        "id",
-        "type"
+      "allOf": [
+        { "required": [ "id", "type" ]},
+        { "oneOf": [
+          { "required": [ "value_number" ] },
+          { "required": [ "value_date" ] },
+          { "required": [ "value_local_constant" ] },
+          { "required": [ "question" ] }
+        ]}
       ],
       "properties": {
         "_comment": {
@@ -127,6 +132,9 @@
         "value_number": {
           "type": "integer"
         },
+        "value_date": {
+          "type": "string"
+        },
         "offset": {
           "type": "integer"
         },
```

Reply:
Ah okay, cool.

Well, maybe we should just leave it in the code because we could add more advanced checks later that relate to the dependent item. For example if you are saying "enable when question X is > 10", we could validate that question X is in fact numeric. _That_ certainly isn't expressible in JSON Schema, I don't think.

Reply:
Sounds good, seems fine to stick with code for now. Probably is possible to express it in json schema but it could get kind of ugly and hard to maintain.